protocol/src/lib.rs

```diff
@@ -140,6 +142,122 @@ impl CrucibleEncoder {
     pub fn new() -> Self {
         CrucibleEncoder {}
     }
+
+    fn serialized_size<T: serde::Serialize>(
+        m: T,
+    ) -> Result<usize, anyhow::Error> {
+        let serialized_len: usize = bincode::serialized_size(&m)? as usize;
+        let len = serialized_len + 4;
+
+        Ok(len)
+    }
+
+    fn a_write(bs: usize) -> Write {
+        Write {
+            eid: 1,
+            offset: Block::new(1, bs.trailing_zeros()),
+            data: {
+                let sz = bs;
+                let mut data = Vec::with_capacity(sz);
+                data.resize(sz, 1);
+                bytes::Bytes::from(data)
+            },
+            nonce: Some(vec![0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]),
+            tag: Some(vec![0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]),
+        }
+    }
+
+    /*
+     * Binary search to find the maximum number of blocks we can send.
+     *
+     * Attempts at deterministically computing the number of blocks
+     * repeatedly failed, so binary search instead.
```

Comment:
Let's just add a little more to the comment here indicating that we are figuring out the max size that the tokio encode/decode supports